### DISCUSSION TITLE
Neper's way of specifying uniform distribution of orientation

### DISCUSSION THREAD

**Discussion by abarun22** [neperfepx/neper#710] [Q&A] [answered]
Hi,
I was tring to check how neper interprets and output the crystal orientation when distributed uniformly for a cuboid model with 10 cells and received the following in the tessallation file.
neper -T -n 10 -oricrysym cubic -ori uniform
```
  *ori
   rodrigues:active
  -0.053259993854    0.134448018624   -0.413853985713
   0.120897860748   -0.279070815081    0.225788708134
  -0.129715498444   -0.030385008250    0.088549530280
   0.264522980004    0.055285847538    0.089727757152
  -0.343973250534    0.205798397660   -0.144724726195
  -0.341319297699    0.232199945700    0.321272702230
   0.023308595029    0.316689612418   -0.031864499903
  -0.324847609454   -0.275816079921    0.351848914217
  -0.319001973493   -0.260950591764   -0.162627413475
   0.087257612216   -0.124783703851   -0.196573393203
```
I was thinking that neper should print same values for all the crystals as all of them are rotated with a single euler angle. A random distribution will assign a specific orientation for each crystal and thererfore i am struggling to understand how uniform values are different from the random ones. Please could someone explain the rationale behind the calculation of these distrubutions?
Kind regards,
Arun
Kind regards,
Arun

**Comment by rquey** [comment, marked as answer]
I don't understand what you mean by
> I was thinking that neper should print same values for all the crystals as all of them are rotated with a single euler angle.

Anyway, see https://neper.info/papers.html#id3 for a description of `-ori uniform`.

**Reply by abarun22** [reply]
Hi Romaine,
Thanks for the paper. I see that in the case of uniform orientations there will never be any clustering of orientations and all of the crystal grains assume nearly a uniform level of orientations. I am now wondering whether the seeds has any relevance here as the picture in this link (https://neper.info/generation.html) talks about the density, which i am unsure of. Does the uniformity also includes distribution of seeds in the relevant length scale?
Kind regards,
Arun

**Reply by rquey** [reply]
No, the distributions of orientations and seeds are completely independent.
